<launch>
    <!-- person_follower -->
    <arg name="use_omni"                    default="false"/>
    <arg name="use_rotate"                  default="true"/>
    <arg name="should_output_marker"        default="true"/>
    <arg name="tracking_method"             default="2"/>
    <!-- Particle Filter base : 1, Particle Filter + Support Vector Data Description : 2 -->

    <!-- rviz -->
    <arg name="rviz"                    default="true"/>
    <arg name="rviz_cfg"                default="$(find person_follower3)/config/rviz/person_follower3.rviz"/>
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_cfg)"/>

    <!-- ssd_node -->
    <include file="$(find person_follower3)/launch/include/object_detect.launch.xml"/>

    <group ns = "person_follower">
        <node pkg="person_follower3" type="person_follower3" name="person_follower3" output="screen">
            <param name="load_model_width_path" type="str" value="$(find svdd_leg_tracker)/model/leg_girth_width.txt"/>
            <param name="load_model_depth_path" type="str" value="$(find svdd_leg_tracker)/model/leg_girth_depth.txt"/>
            <remap from="/person_tracker/follow_position" to="/follow_position" />
            <rosparam file="$(find person_tracking3)/param/topic_params.yaml" command="load" />
            <rosparam file="$(find person_tracking3)/param/particle_person_tracker_params.yaml" command="load" />
            <rosparam file="$(find person_tracking3)/param/svdd_leg_tracker_params.yaml" command="load" />
            <rosparam file="$(find person_tracking3)/param/sensor_rotator_params.yaml" command="load" />
            <rosparam file="$(find follow_path_generator3)/param/dwa_params.yaml" command="load" />
            <rosparam file="$(find follow_path_generator3)/param/path_generator_params.yaml" command="load" />

            <param name="use_omni"              type="bool"     value="$(arg use_omni)"/>
            <param name="use_rotate"            type="bool"     value="$(arg use_rotate)"/>
            <param name="tracking_method"       type="int"      value="$(arg tracking_method)"/>
            <param name="should_output_marker" type="bool"      value="$(arg should_output_marker)"/>
            <param name="is_display_path"      type="bool"      value="$(arg should_output_marker)"/>
            <param name="is_display_all_path"  type="bool"      value="$(arg should_output_marker)"/>

        </node>
    </group>
</launch>